joe/joe.py

```diff
@@ -37,9 +37,17 @@ def _get_data_dir(path):
 
 # Where all gitignore files are
 DATA_DIR = _get_data_dir('*.gitignore')
+
+def _walk_gitignores():
+  '''Recurse over the data directory and return all .gitignore file names'''
+  l = []
+  for root, subFolders, files in os.walk(DATA_DIR):
+    l += [f.replace('.gitignore', '') for f in files if '.gitignore' in f]
```

Comment:
What if my file is called `dontgitignorethisplz`. A safer approach will be this:

    l += [f.replace('.gitignore', '') for f in files if 'f.endswith(.gitignore)']